src/kernel/thread.c

```diff
@@ -188,7 +188,9 @@ void doReplyTransfer(tcb_t *sender, tcb_t *receiver, cte_t *slot, bool_t grant)
                 current_fault = seL4_Fault_Timeout_new(receiver->tcbSchedContext->scBadge);
                 handleTimeout(receiver);
             } else {
-                postpone(receiver->tcbSchedContext);
+                if(thread_state_get_tcbInReleaseQueue(receiver->tcbState) == false){
```

Comment:
Maybe avoid comparing to "false" explicitly:
```suggestion
                if(!thread_state_get_tcbInReleaseQueue(receiver->tcbState)){
```

Reply:
sure